src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -181,10 +187,17 @@ class Diagram extends Component<Props> {
   componentDidMount() {
     this.props.fetchFlows()
     this.props.fetchPrompts()
-    this.setState({ currentLang: this.props.contentLang })
     this.props.fetchContentCategories()
     ReactDOM.findDOMNode(this.diagramWidget).addEventListener('click', this.onDiagramClick)
     document.getElementById('diagramContainer').addEventListener('keydown', this.onKeyDown)
+
+    let expandedNodes
+    try {
+      expandedNodes = JSON.parse(storage.get(EXPANDED_NODES_KEY) || '[]')
+    } catch (error) {
+      expandedNodes = []
+    }
```

Comment:
Being a bit picky; can you move that to a separate method ? there's already a lot of stuff done in the componentDidMount. You can then get rid of your placeholder variable